File: scripts/scratch/check_db_variable.ts
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
import path from 'path';

dotenv.config({ path: path.resolve(process.cwd(), '.env') });
dotenv.config({ path: path.resolve(process.cwd(), '.env.local') });

const supabaseUrl = process.env.VITE_SUPABASE_URL || process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error("[-] Supabase URL or Key not found.");
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function check() {
  const { data, error } = await supabase
    .from('variable_products')
    .select('*')
    .in('sub_type', ['term_pure', 'term_ceo', 'variable_term'])
    .order('id');
    
  if (error) {
    console.error("Error fetching variable products:", error);
    process.exit(1);
  }
  
  console.log(`Total term/ceo/variable_term products in DB: ${data.length}`);
  data.forEach((r) => {
    console.log(`ID: ${r.id} | Company: ${r.company} | Name: ${r.product_name} | SubType: ${r.sub_type} | Male40: ${r.male_premium_40} | Female40: ${r.female_premium_40} | Rate: ${r.declared_rate}`);
  });
}

check();

File: scripts/scratch/check_db_variable.py
import os
import psycopg2
from dotenv import load_dotenv

load_dotenv(".env.local")
load_dotenv(".env")

db_url = os.environ.get("DATABASE_URL")
if not db_url:
    db_url = "postgresql://postgres.wfkxwztxpugakusynhpx:rlaghddlf0411*@aws-0-ap-southeast-1.pooler.supabase.com:5432/postgres"

print("Using DB_URL:", db_url)

try:
    conn = psycopg2.connect(db_url)
    cur = conn.cursor()
    
    cur.execute("SELECT id, company, product_name, sub_type, male_premium_40, female_premium_40, declared_rate, business_fee FROM public.variable_products ORDER BY id;")
    rows = cur.fetchall()
    
    print(f"Total products in variable_products: {len(rows)}")
    for r in rows:
        print(f"ID: {r[0]} | Company: {r[1]} | Name: {r[2]} | SubType: {r[3]} | Male40: {r[4]} | Female40: {r[5]} | Rate: {r[6]} | Fee: {r[7]}")
        
    cur.close()
    conn.close()
except Exception as e:
    print(f"Error checking DB: {e}")
